src/main/java/com/google/api/generator/gapic/composer/ClassComposer.java

```diff
@@ -22,5 +22,7 @@
 
 public interface ClassComposer {
   GapicClass generate(
+      // TODO(miraleung): clean up the hierarchy to avoid pass another parameter (resourceNames is
```

Comment:
As already mentioned [here](https://github.com/googleapis/gapic-generator-java/pull/512#discussion_r528046098), please remove `resourceNames` from this interface and have the class(es) that need it not inherit from ClassComposer. The TODO on me should go into those classes, not this one.